### PR TITLE
Update voiceStateUpdate.ts

### LINKED ISSUE #854: [FEAT] - More Voice Events
### What feature would you like to see?

Some more voice states events

- **voice mute** - when a member has been voice muted (self mute or server mute)
- **voice unmute** - when a member has been voice unmuted (self unmute or server unmute)

- **voice deaf** - when a member has been voice deafed (selfDeaf or serverDeaf)
- **voice undeaf** - when a member has been voice undeafed (self Undeaf or server Undeaf)

### Additional information

Actually I would like to add this Features, if you think this could be useful

### PR DESCRIPTION
**Pull request by L0SER8228**
# Pull Request

## Description

Added more voice update States
Voice mute, unmute, deaf, undeaf

## Related Issue

Closes #854 

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [x] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/events/voiceEvents/voiceStateUpdate.ts [hunk 1]
```diff
@@ -50,6 +46,38 @@ export const voiceStateUpdate = async (
...
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${oldState.selfMute ? "self unmuted" : "server unmuted"}.`
+      );
+      voiceEmbed.setColor(Becca.colours.warning);
+    }
+
+    if (!oldState.deaf && newState.deaf) {
+      voiceEmbed.setTitle("Member voice deafed!");
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${newState.selfDeaf ? "self deafed" : "server deafed"}.`
```
Comment (nhcarrigan) [inline src/events/voiceEvents/voiceStateUpdate.ts:68]:
```suggestion
      voiceEmbed.setTitle("Member voice deafened!");
      voiceEmbed.setDescription(
        `<@!${newState.id}> has been voice ${newState.selfDeaf ? "self deafened" : "server deafened"}.`
```

File: src/events/voiceEvents/voiceStateUpdate.ts [hunk 2]
```diff
@@ -50,6 +46,38 @@ export const voiceStateUpdate = async (
...
+      );
+      voiceEmbed.setColor(Becca.colours.warning);
+    }
+
+    if (oldState.deaf && !newState.deaf) {
+      voiceEmbed.setTitle("Member voice undeafed!");
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${oldState.selfDeaf ? "self undeafed" : "server undeafed"}.`
```
Comment (nhcarrigan) [inline src/events/voiceEvents/voiceStateUpdate.ts:76]:
```suggestion
      voiceEmbed.setTitle("Member voice undeafened!");
      voiceEmbed.setDescription(
        `<@!${newState.id}> has been voice ${oldState.selfDeaf ? "self undeafened" : "server undeafened"}.`
```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

**Comment by L0SER8228** [comment]
Ok, I'll update 👍

**Comment by L0SER8228** [comment]
@nhcarrigan add `hactoberfest` level

File: src/events/voiceEvents/voiceStateUpdate.ts [hunk 3]
```diff
@@ -50,6 +46,38 @@ export const voiceStateUpdate = async (
       voiceEmbed.setColor(Becca.colours.success);
     }
 
+    if (!oldState.mute && newState.mute) {
+      voiceEmbed.setTitle("Member voice muted!");
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${newState.selfMute ? "self muted" : "server muted"}.`
```
Comment (nhcarrigan) [inline src/events/voiceEvents/voiceStateUpdate.ts:52]:
```suggestion
        `<@!${newState.id}> has been ${newState.selfMute ? "self muted" : "server muted"}.`
```

File: src/events/voiceEvents/voiceStateUpdate.ts [hunk 4]
```diff
@@ -50,6 +46,38 @@ export const voiceStateUpdate = async (
...
+      );
+      voiceEmbed.setColor(Becca.colours.error);
+    }
+
+    if (oldState.mute && !newState.mute) {
+      voiceEmbed.setTitle("Member voice unmuted!");
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${oldState.selfMute ? "self unmuted" : "server unmuted"}.`
```
Comment (nhcarrigan) [inline src/events/voiceEvents/voiceStateUpdate.ts:60]:
```suggestion
        `<@!${newState.id}> has been ${oldState.selfMute ? "self unmuted" : "server unmuted"}.`
```

File: src/events/voiceEvents/voiceStateUpdate.ts [hunk 5]
```diff
@@ -50,6 +46,38 @@ export const voiceStateUpdate = async (
...
+      );
+      voiceEmbed.setColor(Becca.colours.success);
+    }
+
+    if (!oldState.deaf && newState.deaf) {
+      voiceEmbed.setTitle("Member voice deafened!");
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${newState.selfDeaf ? "self deafened" : "server deafened"}.`
```
Comment (nhcarrigan) [inline src/events/voiceEvents/voiceStateUpdate.ts:68]:
```suggestion
        `<@!${newState.id}> has been ${newState.selfDeaf ? "self deafened" : "server deafened"}.`
```

File: src/events/voiceEvents/voiceStateUpdate.ts [hunk 6]
```diff
@@ -50,6 +46,38 @@ export const voiceStateUpdate = async (
...
+      );
+      voiceEmbed.setColor(Becca.colours.error);
+    }
+
+    if (oldState.deaf && !newState.deaf) {
+      voiceEmbed.setTitle("Member voice undeafened!");
+      voiceEmbed.setDescription(
+        `<@!${newState.id}> has been voice ${oldState.selfDeaf ? "self undeafened" : "server undeafened"}.`
```
Comment (nhcarrigan) [inline src/events/voiceEvents/voiceStateUpdate.ts:76]:
```suggestion
        `<@!${newState.id}> has been ${oldState.selfDeaf ? "self undeafened" : "server undeafened"}.`
```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
This is looking great - the logic is solid. I just have one small nitpick about the wording and then this will be good to go.

**Comment by nhcarrigan** [comment]
Looks like a couple of lint issues. @L0SER8228 I can push a commit to fix them if you'd like.

**Comment by L0SER8228** [comment]
Ok

**Review by nhcarrigan — APPROVED** [review]
This is good to go. Thanks for all the work on this!

**Comment by L0SER8228** [comment]
You haven't added hactoberfest level on it

**Comment by nhcarrigan** [comment]
It doesn't need a label. PRs count when approved/merged on a repository that has the `hacktoberfest` topic. This one has that topic. 🙂

**Comment by nhcarrigan** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

**Pull request merged**